File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Player.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "players", schema = "public")
public class Player {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "player_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "person_id")
    private Person person;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_type_id")
    private GameType gameType;

    @Column(name = "height", length = Integer.MAX_VALUE)
    private String height;

    @Column(name = "weight", length = Integer.MAX_VALUE)
    private String weight;

    @Column(name = "college", length = Integer.MAX_VALUE)
    private String college;

    @Column(name = "draft_info", length = Integer.MAX_VALUE)
    private String draftInfo;

    @Column(name = "player_status", length = Integer.MAX_VALUE)
    private String playerStatus;

    @Column(name = "jersey_number", length = Integer.MAX_VALUE)
    private String jerseyNumber;

    @Column(name = "player_position", length = Integer.MAX_VALUE)
    private String playerPosition;

    @Column(name = "experience", length = Integer.MAX_VALUE)
    private String experience;

    @Column(name = "birthplace", length = Integer.MAX_VALUE)
    private String birthplace;

}

File: umbrella-insurance-react-ui/src/models/people/players/v1/Player.ts
import { Person } from "../../v1/Person";
import { GameType } from "../../../gameTypes/v1/GameType";

export class Player {
    id?: number;
    gameType?: GameType;
    height?: string;
    weight?: string;
    college?: string;
    draftInfo?: string;
    playerStatus?: string;
    jerseyNumber?: string;
    playerPosition?: string;
    experience?: string;
    birthPlace?: string;
    person?: Person;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.person) {
                this.person = new Person(obj.person);
            }
            if(obj?.gameType) {
                this.gameType = new GameType(obj.gameType);
            }
            if(obj?.height) {
                this.height = obj.height;
            }
            if(obj?.weight) {
                this.weight = obj.weight;
            }
            if(obj?.college) {
                this.college = obj.college;
            }
            if(obj?.draftInfo) {
                this.draftInfo = obj.draftInfo;
            }
            if(obj?.playerStatus) {
                this.playerStatus = obj.playerStatus;
            }
            if(obj?.jerseyNumber) {
                this.jerseyNumber = obj.jerseyNumber;
            }
            if(obj?.playerPosition) {
                this.playerPosition = obj.playerPosition
            }
            if(obj?.experience) {
                this.experience = obj.experience;
            }
            if(obj?.birthPlace) {
                this.birthPlace = obj.birthPlace;
            }
        }
    }
}